include/deal.II/base/bounding_box.h

```diff
@@ -77,14 +77,34 @@ class BoundingBox
   const std::pair<Point<spacedim,Number>,Point<spacedim,Number>> &get_boundary_points () const;
 
   /**
+   * Check if the current object and @p other_bbox are neighbours i.e. if the boxes
+   * have dimension spacedim, check if their insersection is non empty
+   *
+   * Return values:
+   * - 0 : not neighbours
```

Comment:
instead of assigning meaning to each case, use enumerator with those options. Make sure you have tests which cover all 4 cases.